### PR TITLE
[BB-933] Add alternative implementation for retrieving XBlock icon

### PR DESCRIPTION
**Pull request by Agrendalath**
Please see the instructions in the following PR: edx/edx-platform#21433

### CODE REVIEW

**Review by viadanna — APPROVED** [review]
Good to go 👍

- [x] I tested this on my local devstack.
- [x] I read through the code.
- [ ] ~I checked for accessibility issues.~
- [ ] ~Includes documentation.~

**Comment by bradenmacdonald** [comment]
@Agrendalath Is this waiting for another review, or can it be merged?

**Comment by Agrendalath** [comment]
@bradenmacdonald, I merged this, but without edx/edx-platform#21433 it won't work anyway.

**Pull request merged**
